allensdk/brain_observatory/behavior/trial_masks.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import pandas as pd
+
+def trial_types(trials, trial_types):
+    """ only include trials of certain trial types
+
+    Parameters
+    ----------
+    trials : pandas DataFrame
+        dataframe of trials
+    trial_types : list or other iterator
+
+
+    Returns
+    -------
+    mask : pandas Series of booleans, indexed to trials DataFrame
+
+    """
+
+    if trial_types is not None and len(trial_types) > 0:
+        return trials['trial_type'].isin(trial_types)
+    else:
+        return pd.Series(np.ones((len(trials), ), dtype=bool), 
```

Comment:
might want this to be:
```
return pd.Series(True, index=trials.index, name="trial_type")
```

reason is the current implementation fails on a non-default index, which someone could get by subselecting rows from trials:
```
>>> import pandas as pd
>>> import numpy as np
>>> df = pd.DataFrame({"a": [0, 1, 2, 3, 4]}, index=pd.Index([0, 10, 40, 50, 60]))
>>> df
    a
0   0
10  1
40  2
50  3
60  4
>>> all_true = pd.Series(np.ones((len(df), ), dtype=bool), name="true!")
>>> all_true
0    True
1    True
2    True
3    True
4    True
>>> df[all_true]
pandas.core.indexing.IndexingError: Unalignable boolean Series provided as indexer (index of the boolean Series and of the indexed object do not match).
```

vs

```
>>> all_true_again = pd.Series(True, index=df.index, name="true!")
>>> all_true_again
0     True
10    True
40    True
50    True
60    True
Name: true!, dtype: bool
>>> df[all_true_again]
    a
0   0
10  1
40  2
50  3
60  4

```

Reply:
Good catch, the original code just returned a numpy array of `trials` length so I was trying to standardize outputs. I'll update.

Reply:
Completed with tests

Reply:
LGTM